string type_name
float64[] dimensions
